api_client/python/timesketch_api_client/scenario.py

```diff
@@ -55,8 +58,67 @@ def name(self):
         Returns:
             Scenario name as string.
         """
+        if self._name:
+            return self._name
         scenario = self.lazyload_data()
-        return scenario["objects"][0]["name"]
+        self._name = scenario["objects"][0]["name"]
+        return self._name
+
+    @property
+    def display_name(self):
+        """Property that returns the scenario display name."""
+        if self._display_name:
+            return self._display_name
+        scenario = self.lazyload_data()
+        self._display_name = scenario["objects"][0]["display_name"]
+        return self._display_name
+
+    @display_name.setter
+    def display_name(self, value):
+        """Sets the display name of the scenario."""
+        resource_url = f"{self.api.api_root}/{self.resource_uri}"
+        data = {"scenario_name": value}
+        response = self.api.session.post(resource_url, json=data)
+        self._data = error.get_response_json(response, logger)
+        updated_object = self._data["objects"][0]
+        self._display_name = updated_object["display_name"]
+        self._name = updated_object["name"]
+
+    def set_status(self, status):
+        """Sets the status of the scenario.
+
+        Args:
+            status (str): The new status for the scenario.
```

Comment:
I assume the status of a scenario has one of fixed set of values to choose from, I would suggest we enforce that in the setter so we do not end with scenarios in `status: weird` situations.

Reply:
Good catch. We don't use status or scenarios yet. So I'll remove the set_status for scenarios from the API client.

Once we add status for Scenarios we should do the validation of valid status values on the API Endpoint though and not in the client.